src/vocgui/migrations/0037_unrelease_invalid_trainingsets.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 3.2.12 on 2022-03-12 18:27
+from django.conf import settings
+from django.db import migrations
+from django.db.models import Count, Q
+
+
+# pylint: disable=unused-argument
+def unrelease_invalid_trainingsets(apps, schema_editor):
+    """
+    Unrelease existing invalid trainingsets in the database
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+    """
+    # Get model instead of importing to support historic migration even if the TrainingSet gets changed or deleted
+    TrainingSet = apps.get_model("vocgui", "TrainingSet")
+    # Get queryset of all invalid trainingsets in the database
+    queryset = TrainingSet.objects.annotate(
+        document_count=Count(
+            "documents",
+            filter=Q(documents__document_image__confirmed=True),
+            distinct=True,
+        )
+    ).filter(document_count__lt=settings.TRAININGSET_MIN_DOCS, released=True)
+    # Store ids for info message
+    ids = [ts.id for ts in queryset]
+    # Unrelease training sets
+    queryset.update(released=False)
+    # Print info message
+    if ids:
+        print(
+            f"\n    ↳ Unreleased training sets with ids {ids} because they contain less "
+            f"than {settings.TRAININGSET_MIN_DOCS} vocabulary words with confirmed images...",
+            end="",
+        )
```

Comment:
> I wasn't able to test the migration, but the rest seems to work as expected +1

Since the migration changes the production database and cannot be reversed, probably somebody else should have a look, too, and probably we should test it on lunes-test before deploying to production.

Reply:
I took a look and everything seems to work as expected :-)